File: hanoi.js
const hanoi = function (n, A, B, C) {
    if (n === 1) {
        console.log(`${A} --> ${C}`);

    } else {
        // 将第一个塔的圆盘移动到第二个塔
        hanoi(n - 1, A, C, B);
        // 将第一个塔的圆盘移动到第三个塔
        console.log(`${A} --> ${C}`);
        // 将第二个塔移动到第三个塔
        hanoi(n - 1, B, A, C);
    }
};
hanoi(2, "塔1", "塔2", "塔3");

File: 算法/李明杰数据结构/算法刷题/经典题目/Hanoi.java
package 算法刷题.经典题目;

// 汉诺塔
public class Hanoi {
    private static void move(int num, String from, String to) {
        System.out.println("将" + num + "号盘子从" + from + "挪到" + to);
    }

    public static void hanoi(int n, String p1, String p2, String p3) {
        if (n == 1) {
            // 将一号盘子从A挪到C
            move(1, p1, p3);
            return;
        }
        // 将n-1个盘子从A挪到B
        hanoi(n - 1, p1, p3, p2);
        // 将n号盘子从A挪到C
        move(n, p1, p3);
        // 将n-1个盘子从B挪到C
        hanoi(n - 1, p2, p1, p3);
    }

    public static void main(String[] args) {
        hanoi(3, "A", "B", "C");
    }
}
